internal/kgateway/utils/sanitize.go

```diff
@@ -23,3 +24,59 @@ func SanitizeForEnvoy(ctx context.Context, resourceName, resourceTypeName string
 	}
 	return resourceName
 }
+
+// sanitizeCookieName ensures the cookie name is valid per RFC 6265 and Gateway API conventions.
```

Comment:
nit: Match the function name.

```suggestion
// SanitizeCookieName ensures the cookie name is valid per RFC 6265 and Gateway API conventions.
```